### PR TITLE
Avoid uploading glucose data to Remote Services when disabled

### PR DESCRIPTION
**Pull request by bill-foreflight**
This is a potential fix for https://github.com/LoopKit/Loop/issues/1642. In the master branch we were only uploading Glucose data when `CGMManager.shouldSyncToRemoteService` returned true. That feature was lost in dev with subsequent refactors.

This change will respect that value but checking just before a glucose upload is performed. I'm not that familiar with the history of this setting so I may be missing the bigger picture to how this is intended to work.

### CODE REVIEW

File: Loop/Managers/RemoteDataServicesManager.swift [hunk 1]
```diff
@@ -333,6 +335,11 @@ extension RemoteDataServicesManager {
     }
```
Comment (bill-foreflight) [inline Loop/Managers/RemoteDataServicesManager.swift:336]: This is the common method I see that uploads glucose data.

File: Loop/Managers/RemoteDataServicesManager.swift [hunk 2]
```diff
@@ -24,6 +24,8 @@ enum RemoteDataType: String {
 final class RemoteDataServicesManager {
 
     public typealias RawState = [String: Any]
```
Comment (bill-foreflight) [inline Loop/Managers/RemoteDataServicesManager.swift:26]: There was not a reference to cgmManager here so I added a delegate to capture that.

File: Loop/Managers/DeviceDataManager.swift [hunk 3]
```diff
@@ -1530,6 +1531,14 @@ struct CancelTempBasalFailedError: LocalizedError {
     }
 }
 
+//MARK: - RemoteDataServicesManagerDelegate protocol conformance
+
+extension DeviceDataManager : RemoteDataServicesManagerDelegate {
+    var shouldSyncToRemoteService: Bool {
+        return cgmManager?.shouldSyncToRemoteService ?? false
```
Comment (ps2) [inline Loop/Managers/DeviceDataManager.swift:1538]: If we don't have a cgmmanager, but do have glucose values for some reason (unlikely, but possible), then we should probably still upload them.

Comment (bill-foreflight) [reply Loop/Managers/DeviceDataManager.swift]: Good point - done.

**Pull request merged**